src/cli/cli.cpp

```diff
@@ -5793,6 +5795,14 @@ template <> otError Interpreter::Process<Cmd("radiofilter")>(Arg aArgs[])
 }
 #endif
 
+template <> otError Interpreter::Process<Cmd("randombootid")>(Arg aArgs[])
+{
+    static const uint32_t bootid = otRandomNonCryptoGetUint32();
+    OT_UNUSED_VARIABLE(aArgs);
+    OutputLine("%08" PRIx32, bootid);
```

Comment:
Let's not use `PRIx` macros. They are not always available on all platforms and/or can lead to bugs (incorrect headers being used). 

We can do the following
```suggestion
    OutputLine("%08lx", ToUlong(bootid));
```
We use this patten for other CLI commands and logging.